File: Python/Endpoint Examples/JSON Payload/translated-pdf-text.py
import requests
import json

# By default, we use the US-based API service. This is the primary endpoint for global use.
api_url = "https://api.pdfrest.com"

# For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
# For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
#api_url = "https://eu-api.pdfrest.com"

with open('/path/to/file', 'rb') as f:
    upload_data = f.read()

print("Uploading file...")
upload_response = requests.post(
    url=api_url+'/upload',
    data=upload_data,
    headers={
        'Content-Type': 'application/octet-stream',
        'Content-Filename': 'file.pdf',
        'API-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
    }
)

print("Upload response status code: " + str(upload_response.status_code))

if upload_response.ok:
    upload_response_json = upload_response.json()
    print(json.dumps(upload_response_json, indent=2))

    uploaded_id = upload_response_json['files'][0]['id']
    translate_data = {
        "id": uploaded_id,
        # Translates text to American English. Format the output_language as a 2-3 character ISO 639 code, optionally with a region/script (e.g., 'en', 'es', 'zh-Hant', 'eng-US').
        "output_language": "en-US",
    }
    print(json.dumps(translate_data, indent=2))

    print("Processing file...")
    translate_response = requests.post(
        url=api_url+'/translated-pdf-text',
        data=json.dumps(translate_data),
        headers={
            'Content-Type': 'application/json',
            'API-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'
        }
    )

    print("Processing response status code: " + str(translate_response.status_code))
    if translate_response.ok:
        translate_response_json = translate_response.json()
        print(json.dumps(translate_response_json, indent=2))
    else:
        print(translate_response.text)
else:
    print(upload_response.text)

File: JavaScript/Endpoint Examples/JSON Payload/translated-pdf-text.js
var axios = require("axios");
var fs = require("fs");

// By default, we use the US-based API service. This is the primary endpoint for global use.
var apiUrl = "https://api.pdfrest.com";

/* For GDPR compliance and enhanced performance for European users, you can switch to the EU-based service by uncommenting the URL below.
 * For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work
 */
//var apiUrl = "https://eu-api.pdfrest.com";

var upload_data = fs.createReadStream("/path/to/file");

var upload_config = {
  method: "post",
  maxBodyLength: Infinity,
  url: apiUrl + "/upload",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
    "Content-Filename": "filename.pdf",
    "Content-Type": "application/octet-stream",
  },
  data: upload_data,
};

// Send upload request
axios(upload_config)
  .then(function (upload_response) {
    console.log("Upload response:");
    console.log(JSON.stringify(upload_response.data, null, 2));

    var uploaded_id = upload_response.data.files[0].id;

    var translate_config = {
      method: "post",
      maxBodyLength: Infinity,
      url: apiUrl + "/translated-pdf-text",
      headers: {
        "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Replace with your API key
        "Content-Type": "application/json",
      },
      data: {
        id: uploaded_id,
        // Translates text to American English. Format the output_language as a 2-3 character ISO 639 code, optionally with a region/script (e.g., 'en', 'es', 'zh-Hant', 'eng-US').
        output_language: "en-US",
      },
    };

    // Send translated-pdf-text request
    axios(translate_config)
      .then(function (translate_response) {
        console.log("Translate response:");
        console.log(JSON.stringify(translate_response.data, null, 2));
      })
      .catch(function (error) {
        console.error("Translate request error:");
        console.error(error.response?.data || error.message);
      });
  })
  .catch(function (error) {
    console.error("Upload request error:");
    console.error(error.response?.data || error.message);
  });

